### DISCUSSION TITLE
Could not resolve to a Repository with the name ORGANISATION/REPO

### DISCUSSION THREAD

**Discussion by raphaelperrin24** [renovatebot/renovate#32243] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Hello

Renovate fails to get tags from repositories in our organization. All repositories are private.
I set en env variable GITHUB_COM_TOKEN

Is there a way to specify some hostRules for all repos in the organization ?

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Datasource unknown error (repository=myorg/devplat-raphael-playground)
"datasource": "github-tags",
"packageName": "myorg/actions-dosomething",
"err": {
  "message": "Could not resolve to a Repository with the name 'myorg/actions-dosomething'.",
  "stack": "Error: Could not resolve to a Repository with the name 'myorg/actions-dosomething'.\n    at GithubGraphqlDatasourceFetcher.doRawQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:136:21)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at GithubGraphqlDatasourceFetcher.doShrinkableQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:201:20)\n    at GithubGraphqlDatasourceFetcher.doPaginatedFetch (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:250:27)\n    at GithubGraphqlDatasourceFetcher.doCachedQuery (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:290:7)\n    at GithubGraphqlDatasourceFetcher.getItems (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:328:17)\n    at Function.query (/usr/local/renovate/lib/util/github/graphql/datasource-fetcher.ts:60:19)\n    at queryTags (/usr/local/renovate/lib/util/github/graphql/index.ts:15:15)\n    at GithubTagsDatasource.getReleases (/usr/local/renovate/lib/modules/datasource/github-tags/index.ts:78:24)\n    at getRegistryReleases (/usr/local/renovate/lib/modules/datasource/index.ts:85:15)\n    at huntRegistries (/usr/local/renovate/lib/modules/datasource/index.ts:137:13)\n    at fetchReleases (/usr/local/renovate/lib/modules/datasource/index.ts:320:15)\n    at lookupUpdates (/usr/local/renovate/lib/workers/repository/process/lookup/index.ts:140:56)\n    at Function.wrap (/usr/local/renovate/lib/util/stats.ts:38:20)\n    at fetchDepUpdates (/usr/local/renovate/lib/workers/repository/process/fetch.ts:58:42)\n    at /usr/local/renovate/lib/workers/repository/process/fetch.ts:104:23\n    at /usr/local/renovate/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:22"
}

```

</details>

**Comment by rarkins** [comment]
How exactly are you running Renovate? e.g. on github.com, and using a PAT or app token, etc?

GITHUB_COM_TOKEN is for people not already on github.com

**Reply by raphaelperrin24** [reply]
Renovate runs in a GitHub action with a GitHub APP token

**Reply by rarkins** [reply]
If these repos are in the same org on github.com then install them into the app and the app token will then have access.

If they are in a different org, or you don't want to install them for some reason, then you need to set one or more hostRules with a specific matchHost like `"matchHost": "https://api.github.com/myorg/", "hostType": "github", "token": "<some PAT>"` and a PAT `token` with sufficient privileges. One rule per org.

**Reply by raphaelperrin24** [reply]
The repos are on the same org and the app is installed on those repos

I'll anyway test addind a matchHost

**Reply by rarkins** [reply]
Then remove any extra PAT or host rule and focus on getting the normal way working

**Reply by raphaelperrin24** [reply]
Still ain't working with same error

here is my config:

>       - name: Run Renovate
>         uses: renovatebot/github-action@21d88b0bf0183abcee15f990011cca090dfc47dd  # v40.1.12
>         with:
>           token: ${{ secrets.GITHUB_TOKEN }}
>         env:
>           RENOVATE_REPOSITORIES: ${{ github.repository }}
>           LOG_LEVEL: "debug"
> 
> 
> 



> {
>   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
>   "extends": [
>     "config:best-practices",
>     ":semanticPrefixFixDepsChoreOthers"
>   ],
>   "schedule": ["every weekend"],
>   "vulnerabilityAlerts": {
>     "enabled": true
>   },
>   "dependencyDashboard": false
> }

**Reply by rarkins** [reply]
Earlier you wrote:

> a GitHub action with a GitHub APP token

and also:

> The repos are on the same org and the app is installed on those repos

What do you mean by "the app"? 

Because based on your last message it looks like you're running it with the GitHub Actions token (which is always scoped to the one repo) and not using any GitHub App token of your own.

**Reply by raphaelperrin24** [reply]
Now with the token from APP

>       - uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
>         id: generate-app-token
>         with:
>           app-id: ${{ secrets.GHA_APP_ID }}
>           private-key: ${{ secrets.GHA_APP_PRIVATE_KEY }}
> 
>       - name: Run Renovate
>         uses: renovatebot/github-action@21d88b0bf0183abcee15f990011cca090dfc47dd  # v40.1.12
>         with:
>           token: ${{ steps.generate-app-token.outputs.token }}
>         env:
>           RENOVATE_REPOSITORIES: ${{ github.repository }}
>           LOG_LEVEL: "debug"

and still the same

**Reply by rarkins** [reply]
Please read the github action docs carefully.

- it already included an instruction NOT to use GITHUB_TOKEN, which would have saved both of us a lot of time
- it includes an example of using an app token which does not match what you've done above

Starting point https://github.com/renovatebot/github-action#example-with-github-app but ideally read the whole thing

**Reply by raphaelperrin24** [reply]
Hi, sorry for the waste of time.

Simply added:
owner: ${{ github.repository_owner }} to the actions/create-github-app-token made it work.

Thanks a lot

**Comment by raphaelperrin24** [comment, marked as answer]
Working setup:

>   - uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
>     id: generate-app-token
>     with:
>       app-id: ${{ secrets.GHA_APP_ID }}
>       private-key: ${{ secrets.GHA_APP_PRIVATE_KEY }}
>       owner: ${{ github.repository_owner }}
> 
>   - name: Run Renovate
>     uses: renovatebot/github-action@21d88b0bf0183abcee15f990011cca090dfc47dd  # v40.1.12
>     with:
>       token: ${{ steps.generate-app-token.outputs.token }}
>     env:
>       RENOVATE_REPOSITORIES: ${{ github.repository }}
>       LOG_LEVEL: "debug"
